keps/sig-node/3008-cri-class-based-resources/README.md

```diff
@@ -0,0 +1,1504 @@
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We would like to add support for QoS-class resources in Kubernetes.
+QoS-class resources can be thought of as non-accountable resources, each of
+which is presented by a set of classes. Being non-accountable means that
+multiple containers can be assigned to the same class. They are also supposed
+to be opaque to the CRI client in the sense that the container runtime takes
+care of configuration and control of the resources and the classes within.
+
+A prime example of a QoS-class resource is Intel RDT (Resource Director
+Technology). RDT is a technology for controlling the cache lines and memory
+bandwidth available to applications. RDT provides a class-based approach for
+QoS control of these shared resources: all processes in the same hardware class
+share a portion of cache lines and memory bandwidth.
+
+We also believe that the Linux Block IO controller (cgroup) should be handled
+as a QoS-class resource on the level of container orchestration. This enables
+configuring I/O scheduler priority and throttling I/O bandwidth per workload.
+Having the support for QoS-class resources in place, it will provide a
+framework for the future, for instance class-based network or memory type
+prioritization.
+
+We have identified the need for both container-level and pod-level QoS-class
+resources as independent concepts. Intel RDT (above) is per-container by
+design because of the hardware implementation (the control/class hiearchy is
+flat). Also, the current support for blockio is container-level only (it is not
+possible to configure pod sandbox-level cgroup parameters). However, we have
+plans for implementing configuration of sandbox-level blockio parameters. Other
+usage for pod sandbox-level QoS-class resources would be communicating the
+Kubernetes Pod QoS class from kubelet to the container runtime.
```

Comment:
This isn't at all easy to understand. From a SIG Docs point of view, I recommend that we explicitly gate graduation to beta on having good concept documentation to explain what these things are.

Also, I recommend avoiding “we” because it's not clear who “we” represents. I'd guess Intel, but I don't really know.

Reply:
I did a major overhaul of the summary, shortening it and adapting in your suggestions. Removed the word "we" (from some other parts of the doc, too)

Reply:
Thanks.